program/src/processor.rs

```diff
@@ -351,6 +355,104 @@ pub fn process_stake_deposit(
     lido.save(accounts.lido)
 }
 
+/// Unstakes from a validator, the funds are moved to the stake defined by the
+/// validator's unstake seed. Caller must be a maintainer.
+pub fn process_unstake(
+    program_id: &Pubkey,
+    amount: Lamports,
+    raw_accounts: &[AccountInfo],
+) -> ProgramResult {
+    let accounts = UnstakeAccountsInfo::try_from_slice(raw_accounts)?;
+    let mut lido = deserialize_lido(program_id, accounts.lido)?;
+    lido.check_maintainer(accounts.maintainer)?;
+    lido.check_stake_authority(program_id, accounts.lido.key, accounts.stake_authority)?;
+    let destination_bump_seed = check_unstake_accounts(program_id, &lido, &accounts)?;
+
+    let validator = lido.validators.get(accounts.validator_vote_account.key)?;
+    let seeds = [
+        &accounts.lido.key.to_bytes(),
+        &accounts.validator_vote_account.key.to_bytes(),
+        VALIDATOR_UNSTAKE_ACCOUNT,
+        &validator.entry.unstake_seeds.end.to_le_bytes()[..],
+        &[destination_bump_seed],
+    ];
+
+    split_stake_account(
+        accounts.lido.key,
+        &lido,
+        &SplitStakeAccounts {
+            source_stake_account: accounts.source_stake_account,
+            destination_stake_account: accounts.destination_stake_account,
+            authority: accounts.stake_authority,
+            system_program: accounts.system_program,
+            stake_program: accounts.stake_program,
+        },
+        amount,
+        &[&seeds],
+    )?;
+
+    let deactivate_stake_instruction = solana_program::stake::instruction::deactivate_stake(
+        accounts.destination_stake_account.key,
+        accounts.stake_authority.key,
+    );
+
+    // Deactivates the stake. After the stake has become inactive, the Lamports
+    // on this stake account need to go back to the reserve account by using
+    // another instruction.
+    invoke_signed(
+        &deactivate_stake_instruction,
+        &[
+            accounts.destination_stake_account.clone(),
+            accounts.sysvar_clock.clone(),
+            accounts.stake_authority.clone(),
+            accounts.stake_program.clone(),
+        ],
+        &[&[
+            &accounts.lido.key.to_bytes(),
+            STAKE_AUTHORITY,
+            &[lido.stake_authority_bump_seed],
+        ]],
+    )?;
+
+    let validator = lido
+        .validators
+        .get_mut(accounts.validator_vote_account.key)?;
+    // Increase the `unstake_accounts_balance` by `amount`.
+    validator.entry.unstake_accounts_balance = (validator.entry.unstake_accounts_balance + amount)?;
+    msg!(
+        "VALIDATOR ACTIVE: {}, STAKE ACCOUNT:{}",
```

Comment:
Was this a temporary debugging print, or do you want to keep it? If you want to keep it, I would add a little more detailed descriptions of the values.

Reply:
Removed, that was a forgotten temporary debugging print.